test/unit/parse.cpp

```diff
@@ -20,6 +20,11 @@ struct parse_test
     void
     run()
     {
+        // issue 497
+        {
+            url_view u = parse_uri_reference("?~").value();
```

Comment:
well if this fails then it will throw, and screw up CI. Use `BOOST_TEST_NO_THROW` or check the `result`.